src/Microsoft.OData.Core/UriParser/Visitors/PathSegmentToContextUrlPathTranslator.cs

```diff
@@ -126,6 +126,17 @@ public override string Translate(PropertySegment segment)
             return "/" + segment.Property.Name;
         }
 
+        /// <summary>
+        /// Translate an AnnotationSegment
+        /// </summary>
+        /// <param name="segment">the segment to Translate</param>
+        /// <returns>Defined by the implementer.</returns>
+        public override string Translate(AnnotationSegment segment)
+        {
+            Debug.Assert(segment != null, "segment != null");
+            return "/" + segment.Term.Name;

```

Comment:
>Name [](start = 38, length = 4)

Maybe FullName ?

Reply:
Good catch; thanks.

---
In reply to: [166083770](https://github.com/OData/odata.net/pull/1063#discussion_r166083770) [](ancestors = 166083770)